content/ros/ROS_Nodes_Topics_Messages/presentation/ros_nodes_topics_messgaes.tex

```diff
@@ -0,0 +1,1560 @@
+        \end{terminal}
+        
+              
+        
+    \end{itemize}
+    
+\end{frame}
+
+
+\subsection{Writing a publisher node in Python}
+
+\begin{frame}[fragile]{Writing a publisher node in Python}
+    \framesubtitle{ROS Nodes, Topics, and Messages}
+    
+    \begin{itemize}
+        \item  Let's extend our previous node and make it publish a String ROS message.
+    \end{itemize}   
+\end{frame}
+ 
+\begin{frame}[fragile]{Writing a publisher node in Python}
+    \framesubtitle{../scripts/02\_simple\_publisher.py}
+    
+
+                   
+    \lstset{language=python,
+        basicstyle=\scriptsize,
+        keywordstyle=\color{blue}\ttfamily,
+        stringstyle=\color{red}\ttfamily,
+        commentstyle=\color{green}\ttfamily,
+        morecomment=[l][\color{magenta}]{\#},
+        showstringspaces=false
+    }
+\begin{lstlisting}
+#!/usr/bin/env python
+
+import rospy
+
+from std_msgs.msg import String
+
+rospy.init_node('talker')
+
+pub = rospy.Publisher('myFirstTopic', String, queue_size=10)
+
+rate = rospy.Rate(1)
+
+my_message = String()
+my_message.data = "Hello there! How are you?"
+
+while not rospy.is_shutdown():
+    pub.publish(my_message)
+    rate.sleep()
+\end{lstlisting}
+\end{frame}
+
+
+
+
+\begin{frame}[fragile]{Writing a publisher node in Python}
+    \framesubtitle{../scripts/02\_simple\_publisher.py}
+    \begin{focus}
+        \centering
+        \fontsize{9}{1} \ttfamily rospy.Publisher({\color{blue}name}, {\color{blue}data\_class}, {\color{blue}queue\_size})
+    \end{focus}
+    
+    \begin{itemize}
+        \item {\ttfamily name}: Name of the topic to publish on.
+        
+        \item {\ttfamily data\_class}: The type of message. It is a ROS message class.
+        
+        \item {\ttfamily queue\_size}: The size of the outgoing message queue.
+        
+    \end{itemize}
+\end{frame}
+
+\begin{frame}[fragile]{Writing a publisher node in Python}
+    \framesubtitle{../scripts/02\_simple\_publisher.py}
+    \begin{focus}
+        
+        \ttfamily rospy.Publisher(\\
+                                 {\color{red}name},\\
+                                 {\color{red}data\_class},\\
+                                 {\color{blue}subscriber\_listener=None}, \\
+                                 {\color{blue}tcp\_nodelay=False},\\
+                                 {\color{blue}latch=False},\\
+                                 {\color{blue}headers=None},\\
+                                 {\color{red}queue\_size=None}\\)
+    \end{focus}
+
+\end{frame}
+
+
+\begin{frame}[fragile]{Writing a publisher node in Python}
+    \framesubtitle{Things to note..}
+    
+    \begin{itemize}
+        \item  ROS messages are implemented as classes.
+        \item  To publish a message you also need to define a \textbf{Publisher} class.
+        \item Most of ROS concepts and functionalities are implemented as classes. This is why understanding OOP helps you understand ROS better.
```

Comment:
Most of ROS concepts and functionalities -> Most ROS functionalities

But I don't agree that understanding OOP helps in understanding ROS *as a framework* (i.e. OOP is just a programming paradigm that people *often* follow when writing ROS programs, but understanding ROS as such is independent of the programming paradigm).

Reply:
done, and I removed that comment about OOP